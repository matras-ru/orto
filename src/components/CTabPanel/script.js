import { selfInstall } from '@/utils';

export default {
    name: 'CTabPanel',

    install(Vue, theme) {
        selfInstall(Vue, theme, this);
    },

    props: {
        name: {
            type: String,
            default: null
        },

        isActive: {
            type: Boolean,
            default: false
        }
    },

    render(h) {
        return h(
            'section',
            {
                staticClass: 'outline-none select-none',
                attrs: {
                    role: 'tabpanel',
                    id: this.name,
                    'aria-labelledby': `tab-${this.name}`,
                    tabindex: '-1',
                    hidden: !this.isActive
                }
            },
            [this.$slots.default]
        );
    }
};
