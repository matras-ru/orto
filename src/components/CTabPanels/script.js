import { selfInstall } from '@/utils';

export default {
    name: 'CTabPanels',

    install(Vue, theme) {
        selfInstall(Vue, theme, this);
    },

    props: {
        value: {
            type: [String, Number],
            default: null
        },

        lazy: {
            type: Boolean,
            default: false
        }
    },

    model: {
        prop: 'value',
        event: 'change'
    },

    render(h) {
        const normalizeTabPanels = this.$slots.default.map(tabPanel => {
            const { name } = tabPanel.componentOptions.propsData;
            const isActive = this.value === name;

            //
            tabPanel.componentOptions.propsData = {
                ...tabPanel.componentOptions.propsData,
                isActive
            };

            return this.lazy ? (isActive ? tabPanel : null) : tabPanel;
        });

        return h('div', {}, [normalizeTabPanels]);
    }
};
