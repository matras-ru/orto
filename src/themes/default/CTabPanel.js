export const base = '';

export default {
    base
};
