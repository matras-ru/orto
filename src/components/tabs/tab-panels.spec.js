/* eslint-env jest */
import { mount } from '@vue/test-utils';

import CTabPanels from './tab-panels';
import CTabPanel from './tab-panel';

describe('CTabPanels', () => {
    it('default', () => {
        const wrapper = mount(CTabPanels);

        expect(wrapper.is('div')).toBe(true);
    });

    it('with panels', () => {
        const wrapper = mount(CTabPanels, {
            slots: { default: [CTabPanel, CTabPanel] }
        });

        expect(wrapper).toBeDefined();
        expect(wrapper.findAll(CTabPanel).length).toBe(2);
    });

    it('panel', () => {
        const wrapper = mount(CTabPanels, {
            slots: { default: [CTabPanel] }
        });

        const panel = wrapper.find(CTabPanel);
        expect(wrapper).toBeDefined();

        expect(panel.attributes('role')).toBeDefined;
        expect(panel.attributes('role')).toBe('tabpanel');

        expect(panel.attributes('id')).toBeDefined;
        expect(panel.attributes('aria-labelledby')).toBeDefined;
    });
});
