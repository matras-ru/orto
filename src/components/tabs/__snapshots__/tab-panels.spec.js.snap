// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTabPanels with panels 1`] = `
<div>
  <section role="tabpanel" aria-labelledby="tab-null"></section>
  <section role="tabpanel" aria-labelledby="tab-null"></section>
</div>
`;
